agent/acs/handler/acs_handler.go

```diff
@@ -194,6 +194,12 @@ func (acsSession *session) Start() error {
 			seelog.Debugf("Received connect to ACS message")
 			// Start a session with ACS
 			acsError := acsSession.startSessionOnce()
+			select {
+			case <-acsSession.ctx.Done():
```

Comment:
why this is required? the below `acsSession.ctx.Done()` should be sufficient right?

Reply:
this is checking if the Done channel is closed, it passes through and continues if it is not closed via the `default:`

Reply:
errr i misunderstood the question. When we receive acsError we go into a whole series of checks and retries that we don't need to do when we are just going to exit. Changing the logic around handling all of that would be a much larger refactor so I have added an exit check right here to short-circuit it.